collection-scripts/gather_audit_logs

```diff
@@ -1,37 +1,24 @@
 #!/bin/bash
-BASE_COLLECTION_PATH="/must-gather"
-AUDIT_LOG_PATH="${BASE_COLLECTION_PATH}/audit_logs/"
-
-### Helper Functions
-function queue() {
-    local TARGET="${1}"
-    shift
-    local LIVE="$(jobs | wc -l)"
-    while [[ "${LIVE}" -ge 45 ]]; do
-        sleep 1
-	LIVE="$(jobs | wc -l)"
-    done
-    echo "${@}"
-    if [[ -n "${FILTER}" ]]; then
-        "${@}" | "${FILTER}" >"${TARGET}" &
-    else
-        "${@}" >"${TARGET}" &
-    fi
-}
-
-# Collect System Audit Logs
-function collect_audit_logs {  ### Takes an input of PATH 
-                               ###  (openshift-apiserver or kube-apiserver)
-
-    echo "WARNING: Collecting one or more audit logs on ALL masters in your cluster. This could take a large amount of time." >&2
-    mkdir -p ${AUDIT_LOG_PATH}/${1}
-    /usr/bin/oc adm node-logs --role=master --path=${1}/ > ${AUDIT_LOG_PATH}/${1}.audit_logs_listing
-    while IFS=$'\n' read -r line; do
-        IFS=' ' read -ra log <<< "${line}"
-	FILTER=gzip queue ${AUDIT_LOG_PATH}/${1}/"${log[0]}"-"${log[1]}".gz /usr/bin/oc adm node-logs "${log[0]}" --path=${1}/"${log[1]}"
-    done < ${AUDIT_LOG_PATH}/${1}.audit_logs_listing
-    echo "INFO: Audit logs for $1 collected."
-}
-
-collect_audit_logs openshift-apiserver
-collect_audit_logs kube-apiserver
+# Downloads the audit.log (and its rotated copies) from
+# /var/logs/{kube-apiserver,openshift-apiserver} on each
+# master node.
+echo "WARNING: Collecting one or more audit logs on ALL masters in your cluster. This could take a large amount of time." >&2
+# the command executed by xargs below expects four parameters:
+# $1 - node path under /var/logs to download
+# $2 - local output path
+# $3 - node name
+# $4 - log file name
+paths=(openshift-apiserver kube-apiserver)
+for path in "${paths[@]}" ; do
+  output_dir="${ARTIFACT_DIR}/audit-logs/$path"
+  mkdir -p "$output_dir"
+  oc adm node-logs --role=master --path="$path" | \
+  tee "${ARTIFACT_DIR}/audit-logs/$path.audit_logs_listing" | \
+  sed "s|^|$path $output_dir |"
+done | \
+xargs --max-args=4 --max-procs=45 bash -c \
+  'echo "INFO: Started downloading $4 from $3";
+  oc adm node-logs $3 --path=$1/$4 | gzip > $2/$3-$4.gz;
```

Comment:
Can you add  `--since ${SINCE_TIMEFRAME}` to this?

Reply:
@sferich888 , I have added the `--since` parameter, but it doesn't seem to actually do anything.

Reply:
The `--since` parameter is only applied to journal logs. It it ignored for `--path` logs, which are simply served by an instance of `http.FileServer`.

Reply:
What other ways can we limit the collection? 

Is an awk script needed? 

`#!/usr/bin/gawk -f BEGIN { starttime = mktime(starttime) endtime = mktime(endtime) } func in_range(n, start, end) { return start <= n && n < end } match($0, /^([0-9]{4})-([0-9]{2})-([0-9]{2})\s/, m) && in_range(mktime(m[1] " " m[2] " " m[3] " 00 00 00"), `

Source: https://askubuntu.com/questions/800769/how-to-get-text-from-range-of-dates-using-grep-sed-in-large-text-file

Reply:
@sferich888 I think I can work something out, but as a followup.

Reply:
In others scripts I see the full path of command, like:/usr/bin/oc